### PR TITLE
8260695: The java.awt.color.ICC_Profile#getData/getData(int) are not thread safe

### PR DESCRIPTION
**Pull request by mrserb**
Both methods are implemented in a similar way.
 1. Requests the size of the profile/tag data
 2. Creates an array of the correct size
 3. Requests the data and copy it to the array

If the data will be changed concurrently between steps 2. and 3. then we will get a mismatch between the array and copied data. 

In the fix, all steps above are merged to just one step - return the data when requested.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260695](https://bugs.openjdk.java.net/browse/JDK-8260695): The java.awt.color.ICC_Profile#getData/getData(int) are not thread safe

### Reviewers
 * [Alexander Zvegintsev](https://openjdk.java.net/census#azvegint) (@azvegint - **Reviewer**) ⚠️ Review applies to a9347ae397a5c34530111a6e4020ea68d8377fc3
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2330/head:pull/2330`
`$ git checkout pull/2330`

### CODE REVIEW

File: src/java.desktop/share/classes/java/awt/color/ICC_Profile.java [hunk 1]
```diff
@@ -1245,25 +1231,12 @@ public void write(OutputStream s) throws IOException {
...
-            /* get the number of bytes needed for this tag */
-            tagSize = mdl.getTagSize(p, tagSignature);
-
-            tagData = new byte[tagSize]; /* get an array for the tag */
-
-            /* get the tag's data */
-            mdl.getTagData(p, tagSignature, tagData);
-        } catch(CMMException c) {
-            tagData = null;
+            return CMSManager.getModule().getTagData(p, tagSignature);
```
Comment (mrserb) [inline src/java.desktop/share/classes/java/awt/color/ICC_Profile.java:1236]: Just one step, the array will be created in the native code under proper synchronization.

File: src/java.desktop/share/native/liblcms/LCMS.c [hunk 2]
```diff
@@ -309,70 +309,46 @@ JNIEXPORT jlong JNICALL Java_sun_java2d_cmm_lcms_LCMS_loadProfileNative
...
+        return NULL;
     }
 
-    // verify java buffer capacity
-    size = (*env)->GetArrayLength(env, data);
-    if (0 >= size || pfSize > (cmsUInt32Number)size) {
-        JNU_ThrowByName(env, "java/awt/color/CMMException",
-                        "Insufficient buffer capacity.");
-        return;
+    jbyteArray data = (*env)->NewByteArray(env, pfSize);
```
Comment (mrserb) [inline src/java.desktop/share/native/liblcms/LCMS.c:330]: Here we create the new array.

File: src/java.desktop/share/classes/sun/java2d/cmm/CMSManager.java [hunk 3]
```diff
@@ -100,33 +100,19 @@ public Profile loadProfile(byte[] data) {
             return p;
         }
 
-        public int getProfileSize(Profile p) {
```
Comment (azvegint) [inline src/java.desktop/share/classes/sun/java2d/cmm/CMSManager.java:103]: Not sure if it is a big loss, but it looks like we are no longer printing profile and tag sizes under CMMTracer after the fix.

**Review by azvegint — APPROVED** [review]

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**